plan/aggregation_push_down.go

```diff
@@ -315,19 +315,23 @@ func (a *aggPushDownSolver) pushAggCrossUnion(agg *Aggregation, unionSchema *exp
 	return newAgg
 }
 
-func (a *aggPushDownSolver) optimize(p LogicalPlan, ctx context.Context, alloc *idAllocator) (LogicalPlan, error) {
+func (a *aggregationOptimizer) optimize(p LogicalPlan, ctx context.Context, alloc *idAllocator) (LogicalPlan, error) {
 	if !ctx.GetSessionVars().AllowAggPushDown {
 		return p, nil
 	}
 	a.ctx = ctx
-	a.alloc = alloc
+	a.allocator = alloc
 	a.aggPushDown(p)
 	return p, nil
 }
 
 // aggPushDown tries to push down aggregate functions to join paths.
-func (a *aggPushDownSolver) aggPushDown(p LogicalPlan) {
+func (a *aggregationOptimizer) aggPushDown(p LogicalPlan) LogicalPlan {
 	if agg, ok := p.(*Aggregation); ok {
+		proj := a.tryToEliminateAggregation(agg)
+		if proj != nil {
+			return proj
```

Comment:
sql like `select count(1) from (select count(1), a, b from t group by a, b) tt group by b;`, if both a and b are key, the inner aggregation won't be eliminated.